incident_reporting/external/google_nbd.py

```diff
@@ -177,12 +177,12 @@ def _get_incidents_back_x_days(
         stored_df = self._get_stored_incidents()
 
         with self.client.context():
+            stored_df = stored_df.filter(
+                pl.col(KEY_REPORTED_DATE) >= date_limit
+            )
+
             # Check if the data frame has any rows.
             if stored_df.shape[0]:
-                stored_df = stored_df.filter(
-                    pl.col(KEY_REPORTED_DATE) >= date_limit
-                )
-
```

Comment:
I was filtering the incidents after I checked to see if there were any appropriate incidents, which meant that once the cache was a month back the application would error. cc: @Nik4002